test/common/filter/config_discovery_impl_test.cc

```diff
@@ -191,8 +191,9 @@ class FilterConfigDiscoveryImplTest : public FilterConfigDiscoveryTestBase {
     }
 
     return filter_config_provider_manager_->createDynamicFilterConfigProvider(
-        config_source, name, server_factory_context_, factory_context_, last_filter_config,
-        getFilterType(), getMatcher());
+        config_source, name, server_factory_context_, factory_context_,
```

Comment:
PR looks good, can you also add an explicit upstream filter integration test?

Reply:
@htuch yes I already created draft (added links in the issue post), but since those tests are going to be a lot of lines of code I prefered to split to multiple PRs rather than mix them.

Reply:
Sure, that's fair as well.